Server: add and support unix listener (UDS) (#18227)
hashicorp/vault: website/content/docs/configuration/listener/unix.mdx

---
layout: docs
page_title: Unix - Listeners - Configuration
description: |-

>   The Unix listener configures Vault to listen on the specified unix domain socket.

Reviewer: ```suggestion
  The Unix listener configures Vault to listen on the specified Unix domain socket.
```

Reviewer: Unix is a brand name, like Nike. But not UNIX because we aren't using the trademark

---

# `unix` Listener

> The unix listener configures Vault to listen on the specified unix domain socket.

Reviewer: ```suggestion
The Unix listener configures Vault to listen on the specified Unix domain socket.
```

```hcl
listener "unix" {
  address = "/run/vault.sock"
}
```

The `listener` stanza may be specified more than once to make Vault listen on
multiple sockets.

## `unix` Listener Parameters

> - `address` `(string: "/run/vault.sock")` – Specifies the address to bind the unix domain socket.

Reviewer: ```suggestion
- `address` `(string: "/run/vault.sock")` – Specifies the address to bind the Unix domain socket.
```

> - `socket_mode` `(string: "")` – When specified, change the access 

Reviewer 2: ```suggestion
- `socket_mode` `(string: "")` – When specified, changes the access 
```

>   permissions and the special mode flags of the unix socket.

Reviewer: ```suggestion
  permissions and the special mode flags of the Unix socket.
```

> - `socket_user` `(string: "")` – When specified, change the user owner of unix socket.

Reviewer 2: ```suggestion
- `socket_user` `(string: "")` – When specified, changes the user owner of the unix socket.
```

> - `socket_user` `(string: "")` – When specified, changes the user owner of the unix socket.

Reviewer: ```suggestion
- `socket_user` `(string: "", <optional>)` – Changes the user owner of the Unix socket.
```

> - `socket_group` `(string: "")` – When specified, change the group owner of unix socket.

Reviewer 2: ```suggestion
- `socket_group` `(string: "")` – When specified, changes the group owner of the unix socket.
```

> - `socket_user` `(string: "")` – When specified, changes the user owner of the unix socket.
> 
> - `socket_group` `(string: "")` – When specified, changes the group owner of the unix socket.

Reviewer: ```suggestion
- `socket_user` `(string: "")` – When specified, changes the user owner of the Unix socket.

- `socket_group` `(string: "")` – When specified, changes the group owner of the Unix socket.
```

> - `socket_group` `(string: "")` – When specified, changes the group owner of the unix socket.

Reviewer: Are these optional?

Reviewer 2: Yes

> - `socket_group` `(string: "")` – When specified, changes the group owner of the unix socket.

Reviewer: ```suggestion
- `socket_group` `(string: "", <optional>)` – Changes the group owner of the unix socket.
```

> - `address` `(string: "/run/vault.sock")` – Specifies the address to bind the unix domain socket.
> 
> - `socket_mode` `(string: "")` – When specified, changes the access
>   permissions and the special mode flags of the unix socket.
> 
> - `socket_user` `(string: "")` – When specified, changes the user owner of the unix socket.
> 
> - `socket_group` `(string: "")` – When specified, changes the group owner of the unix socket.

Reviewer: ```suggestion
- `address` `(string: "/run/vault.sock", <required>)` – Specifies the address to bind the Unix domain socket.

- `socket_mode` `(string: "", <optional>)` – Changes the access
  permissions and the special mode flags of the unix socket.

- `socket_user` `(string: "", <optional>)` – Changes the user owner of the Unix socket.

- `socket_group` `(string: "", <optional>)` – Changes the group owner of the Unix socket.
```

## `unix` Listener Examples

### Listening on Multiple Sockets

This example shows Vault listening on a specified socket, as well as default.

```hcl
listener "unix" {}

listener "unix" {
  address = "/var/run/vault.sock"
}
```

### Listening on Multiple Interfaces

> This example shows Vault listening on tcp localhost, as well as unix socket.

Reviewer 2: ```suggestion
This example shows Vault listening on TCP localhost, as well as unix socket.
```

> This example shows Vault listening on TCP localhost, as well as unix socket.

Reviewer: ```suggestion
This example shows Vault listening on TCP localhost, as well as Unix socket.
```

```hcl
listener "unix" {
  address = "/var/run/vault.sock"
}

listener "tcp" {
  address = "127.0.0.1:8200"
}
```

### Configuring Permissions

> This example shows changeing access permissions and ownership of unix socket.

Reviewer 2: ```suggestion
This example shows changing access permissions and ownership of the unix socket.
```

> This example shows changing access permissions and ownership of the unix socket.

Reviewer: ```suggestion
This example shows changing access permissions and ownership of the Unix socket.
```
